File: ICPC/2024 russia/E.cpp
#include<bits/stdc++.h>

using namespace std;

using ll = long long;

int main(){
    cin.tie(nullptr)->sync_with_stdio(false);
    int n;
    ll m;
    cin >> n >> m;
    vector<ll> a(n);
    for(auto &x:a){
        cin >> x;
    }
    long double tot=0;
    sort(a.rbegin(),a.rend());
    for(ll i=0;i<n;i++){
        m-=a[i]/(i+1);
        ll rem=a[i]%(i+1);
        if(rem>0){
            tot+=(long double)(a[i]%(i+1))/(i+1);
        }
    }
    if(tot-m>=-1e-9){
        cout << "YES\n";
    }else{
        cout << "NO\n";
    }
}

File: ICPC/2024 russia/E.py
n,x=map(int,input().split())
a=[*map(int,input().split())]
tot=0
a.sort()
for i,e in enumerate(a[::-1]):
    tot+=e/(i+1)
if tot-x>=-1e-9:
    print("YES")
else:
    print("NO")